modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -1446,7 +1446,16 @@ def isin_func(df, values):
     abs = Map.register(pandas.DataFrame.abs, dtypes="copy")
     applymap = Map.register(pandas.DataFrame.applymap)
     conj = Map.register(lambda df, *args, **kwargs: pandas.DataFrame(np.conj(df)))
-    convert_dtypes = Map.register(pandas.DataFrame.convert_dtypes)
+
+    def convert_dtypes(self, *args, **kwargs):
+        new_modin_frame = self._modin_frame.apply_full_axis(
+            0,
+            lambda df: df.convert_dtypes(*args, **kwargs),
+            new_index=self._modin_frame._index_cache,
+            new_columns=self._modin_frame._columns_cache,
+        )
+        return self.__constructor__(new_modin_frame)
```

Comment:
Can we use an algebra operator for that purpose? [`Fold` explicitly preserves](https://github.com/modin-project/modin/blob/40df200c67ef93dbbe86a3b15ab01926ff7e3faf/modin/core/dataframe/pandas/dataframe/dataframe.py#L1845-L1846) row lengths and column widths in contrasts with `.apply_full_axis` that does so implicitly and so may no always do so 
```suggestion
    convert_dtypes = Fold.register(pandas.DataFrame.convert_dtypes)
```
(fold with no axis arguments applies across 0 axis)